pennylane_honeywell/device.py

```diff
@@ -128,15 +164,184 @@ def set_api_configs(self):
+
+        Args:
+            access_token (str): access token to save
+            refresh_token (str): refresh token to save (if any)
+        """
+        config = qml.default_config
+
+        config.safe_set(config._config, access_token, *["honeywell", "global", "access_token"])
+
+        if refresh_token is not None:
+            config.safe_set(
+                config._config, refresh_token, *["honeywell", "global", "refresh_token"]
+            )
+
+        directory, _ = os.path.split(qml.default_config._filepath)
+
+        if not directory:
+            directory = config._user_config_dir
+            config._filepath = os.path.join(directory, config._name)
+
+        if not os.path.isdir(directory):
+            os.mkdir(directory)
+
+        with open(qml.default_config._filepath, "w") as f:
+            toml.dump(config._config, f)
+
+    def _login(self):
+        """Login to the HQS service to obtain an access token and a refresh token.
+
+        Returns:
+            tuple: a valid access token and a refresh token
+
+        Raises:
+            ValueError: if no username was provided
+            RequestFailedError: if the request failed
+        """
+
+        if not self._user:
+            raise ValueError("No username for HQS platform found when trying to login.")
+
+        pwd = getpass.getpass(prompt="Enter your Honeywell account password: ")
+        body = {"Content-Type": "application/json", "email": self._user, "password": pwd}
+
+        response = requests.post("https://qapi.honeywell.com/v1/login", json=body)
+
+        if response.status_code == 200:
+            response_json = response.json()
+            access_token = response_json["id-token"]
+            refresh_token = response_json["refresh-token"]
+
+            # Delete the user credential
+            pwd = None
+            return access_token, refresh_token
+
+        raise RequestFailedError(
+            f"Failed to get access token: {self._format_error_message(response)}"
+        )
+
+    @staticmethod
+    def _format_error_message(response):
+        """Formats an error message of an HTTP response.
+
+        Args:
+            response: the HTTP response to format
+
+        Returns:
+            str: a formatted version of the response
+        """
+        body = response.json()
+        status = body.get("status_code", "")
+        code = (body.get("code", ""),)
+        detail = (body.get("detail", ""),)
+        meta = (body.get("meta", ""),)
+        return f"{status} ({code}): {detail} ({meta})"
+
+    def _refresh_access_token(self):
+        """Sends a request to refresh the access token using the stored refresh
+        token.
+
+        Returns:
+            str: a valid access token
+
+        Raises:
+            RequestFailedError: if the request failed
+        """
+        # Refresh the access token using the refresh token
+        body = {"refresh-token": self._refresh_token}
```

Comment:
How come the `"Content-Type": "application/json"` header isn't needed here?

Reply:
Good catch! As per your other suggestion, now added it here. This method was not being used before.